### DISCUSSION TITLE
transmission-qt doesn't respect system application style

### DISCUSSION THREAD

**Discussion by squash0r4ng3** [transmission/transmission#5087] [Q&A]
### What is the issue?

transmission-qt style is stuck on 'Fusion', system uses 'Breeze'. 
Arch, KDE 

### Which application of Transmission?

Qt app on Linux, BSD, etc.

### Which version of Transmission?

4.0.1

**Comment by ckerr** [comment]
Transmission doesn't do anything with Qt theming AFAIK so I don't think this is a bug in the codebase -- converting to discussion.

Could it possibly be that you have different themes set on your system for Qt5 apps vs Qt6 apps?

**Reply by squash0r4ng3** [reply]
I wasn't aware you can do that in KDE, differentiate themes between Qt5 and Qt6 apps.. 
Anyway, Transmission was behaving normal until the 4.0.1 update, which yesterday arrived in the repos, I was previously on the 3.0 ver, again, no complaints before..

**Reply by squash0r4ng3** [reply]
I've downgraded to 3.0 for the time being, but I still consider this a bug in the 4.0.1, it's the only app exhibiting this behavior.

**Reply by ckerr** [reply]
You're the only person to report this so far. If you're going to ignore my questions then there's not a lot of action I can take here.

**Reply by squash0r4ng3** [reply]
Well I answered your question, I don't know how to set in KDE different themes for Qt5 and Qt6 apps. To my knowledge they're all the same.

**Reply by mikedld** [reply]
This is basically the same as Qt 4 -> 5 transition, see e.g. https://stackoverflow.com/questions/17588750/how-to-build-kde-oxygen-plugin-on-qt-5. Seems that KDE has its own style plugin (platform theme) that needs to be built against Qt 6 to work its magic for Qt 6 apps, much like there's qt5ct plugin that doesn't simply work for Qt 6 and qt6ct is required.

**Comment by ckerr** [comment]
https://www.reddit.com/r/archlinux/comments/y4814a/qt_apps_not_using_current_application_style_theme/

> More apps are using QT6 which is not yet supported by KDE (because they are on QT5).

https://www.reddit.com/r/kde/comments/urug5v/guide_to_a_consistent_application_style_in_plasma/ has a nice long writeup on this. Among other things, it says

> There aren't yet many Qt6 apps and many can still be built with Qt5

This is true of Transmission as well; you can rebuild it with Qt5.

**Reply by squash0r4ng3** [reply]
So it's actually Plasma's 'fault'? Not being fully transitioned to Qt6?

**Comment by squash0r4ng3** [comment]
I found somewhat of a theming solution, but it requires using kvantum:
1. install `kvantum`, `qt6ct` 
2. `sudo nano /etc/environment` and add: `QT_QPA_PLATFORMTHEME=qt6ct`
3. relogin, run qt6ct, select a style other than "fusion" (kvantum). I don't have brezee in that list..
Still.. the padding for every UI element is all wrong, triple than normal.
